html/mycakeapp/src/Controller/ReservesController.php

```diff
@@ -161,6 +175,47 @@ public function discount()
 
     public function checkdetail()
     {
+        $memberId = $this->Auth->user('id');
+        $mainKey = [
+            'member_id' => $memberId,
+            'schedule_id' => $_SESSION['schedule_id'],
+            'column_number' => $_SESSION['column_number'],
+            'record_number' => $_SESSION['record_number'],
+        ];
+        $reservationDetails = $this->ReservationDetails->find('ReservationDetails', ['mainKey' => $mainKey]);
+        if (!empty($_SESSION['checkdetail'])) {
+            $mainKey = [
+                'member_id' => $memberId,
+                'schedule_id' => $_SESSION['schedule_id'],
+                'column_number' => $_SESSION['column_number'],
+                'record_number' => $_SESSION['record_number'],
+            ];
+            $reservationDetails = $this->ReservationDetails->find('ReservationDetails', ['mainKey' => $mainKey]);
+            $reservationDetails[0]['is_cancelled'] = 1;
+            if (!($this->ReservationDetails->save($reservationDetails[0]))) {
+                $this->request->session()->delete('seat');
+                return $this->redirect(['controller' => 'error']);
+            }
+            $movieDetails = $this->Schedules->find('MovieDetails', ['schedule_id' => $reservationDetails[0]['schedule_id']]);
+            $_SESSION['detail'] = [
+                'movieTitle' => $movieDetails->movie->name,
+                'date' => $movieDetails->movie->date,
+                'week' => $movieDetails->movie->week,
+                'startTime' => $movieDetails->movie->start_time,
+                'finishTime' => $movieDetails->movie->finish_time,
+                'seatColumn' => $reservationDetails[0]['column_number'],
+                'seatRecord' => $reservationDetails[0]['record_number'],
+                'fee_id' => $reservationDetails[0]['fee_id'],
+                'discount_id' => $reservationDetails[0]['discount_id'],
+            ];
+            $_SESSION['seat'] = [
+                'member_id' => $memberId,
+                'schedule_id' => $reservationDetails[0]['schedule_id'],
+                'column_number' => $reservationDetails[0]['column_number'],
+                'record_number' => $reservationDetails[0]['record_number'],
+            ];
+            $_SESSION['fee'] = $this->Fees->get($_SESSION['detail']['fee_id']);
+        }
         //直接画面遷移の対応(途中でページを遷移したことによりセッションは残っていた場合も直接遷移させない)
         if (!(isset($_SESSION['detail']['discount_id'])) || $this->referer(null, true) !== '/reserves/discount' && $this->referer(null, true) !== '/reserves/checkdetail') {
             $this->request->session()->delete('seat');
```

Comment:
一点修正追加です。今のままだと決済画面から戻るを押すとエラーが出るので
```suggestion
        if (!(isset($_SESSION['detail']['discount_id'])) || $this->referer(null, true) !== '/reserves/discount' && $this->referer(null, true) !== '/reserves/checkdetail' && $this->referer(null, true) !== '/reserves/payment') {
            $this->request->session()->delete('seat');
```
このように変更お願いします！

Reply:
条件文が非常に長くなったので、不正遷移の場合の変数を格納し、if文で値を調べる形へ変更しました。